docs/content/sips/0xx-spin-factors.md

```diff
@@ -0,0 +1,108 @@
+title = "SIP 0XX - Spin Factors"
+template = "main"
+date = "2024-05-20T12:00:00Z"
+
+---
+
+Summary: Refactor Spin Runtime Functionality with "Factors"
+
+Owner: lann.martin@fermyon.com
+
+Created: May 20, 2024
+
+## Background
+
+Spin 1.0 shipped with a mechanism for extending the runtime environment with
+loosely-coupled host functionality called "host components". As Spin has
+evolved, more and more runtime functionality has fallen outside of the scope of
+this mechanism, leading to the code for these features spreading out over the
+Spin codebase. This not only makes it hard to read and understand these features
+but also raises the bar for unfamiliar developers trying to add new features.
+
+Separately, the introduction of the SpinKube project has made it clear that
+there will be (at least) two major embeddings of the Spin runtime environment:
+the Spin CLI (`spin up`) and SpinKube (`containerd-shim-spin`). In order to
+provide better integration into the Kubernetes ecosystem, some runtime feature
+implementations will diverge between these embeddings, making the loose coupling
+of features even more important.
+
+## Proposal
+
+The basic inversion of control approach used by `HostComponent`s will be
+redesigned and expanded into a new system called Spin Factors, where independent
+feature sets are organized into individual "factors". Some of the goals of this
```

Comment:
I understand the term "factor" is intentionally vague, and we'll never really be able to come up with a concise word to describe what a factor is meant to represent. That being said, it would be massively useful if this SIP attempted to clarify what a factor is and perhaps more importantly what a factor is not.

It's hard to make a decision on this refactoring path if we're deciding on adopting a concept that isn't explicitly defined.